packages/contracts/contracts/Unirep.sol

```diff
@@ -344,24 +392,21 @@ contract Unirep is IUnirep, zkSNARKHelper, VerifySignature {
         bytes32 proofNullifier = keccak256(
             abi.encodePacked(publicSignals, proof)
         );
-        address sender = msg.sender;
         verifyProofNullifier(proofNullifier);
-        verifyAttesterSignUp(sender);
-        verifyAttesterIndex(sender, publicSignals[3]);
-        verifyAttesterFee();
 
         if (publicSignals[2] != currentEpoch) revert EpochNotMatch();
-        if (publicSignals[0] > maxEpochKey) revert InvalidEpochKey();
-
-        // Add to the cumulated attesting fee
-        collectedAttestingFee = collectedAttestingFee.add(msg.value);
 
         // attestation of airdrop
         Attestation memory attestation;
-        attestation.attesterId = attesters[msg.sender];
+        attestation.attesterId = publicSignals[3];
         attestation.posRep = airdropAmount[msg.sender];
```

Comment:
should we verify `attesters[msg.sender] == publicSignals[3]`?

Reply:
Yes, but this function is removed in #87